src/Symfony/Component/Finder/Gitignore.php

```diff
@@ -14,120 +14,78 @@
 /**
  * Gitignore matches against text.
  *
+ * @author Michael Voříšek <vorismi3@fel.cvut.cz>
  * @author Ahmed Abdou <mail@ahmd.io>
  */
 class Gitignore
 {
     /**
      * Returns a regexp which is the equivalent of the gitignore pattern.
      *
-     * @return string The regexp
+     * Format specification: https://git-scm.com/docs/gitignore#_pattern_format
      */
     public static function toRegex(string $gitignoreFileContent): string
     {
-        $gitignoreFileContent = preg_replace('/^[^\\\r\n]*#.*/m', '', $gitignoreFileContent);
-        $gitignoreLines = preg_split('/\r\n|\r|\n/', $gitignoreFileContent);
+        $gitignoreFileContent = preg_replace('~(?<!\\\\)#[^\n\r]*~', '', $gitignoreFileContent);
+        $gitignoreLines = preg_split('~\r\n?|\n~', $gitignoreFileContent);
 
-        $positives = [];
-        $negatives = [];
+        $res = self::lineToRegex('');
         foreach ($gitignoreLines as $i => $line) {
-            $line = trim($line);
-            if ('' === $line) {
-                continue;
+            if (\strlen($line) < 2 || '\\' !== substr($line, -2, 1)) {
+                $line = rtrim($line, ' ');
```

Comment:
what's about tabs?

Reply:
`.gitignore` spec says `spaces` explicitly

Reply:
by spec, citing:

> Trailing spaces are ignored unless they are quoted with backslash ("\").

Reply:
I now verified `spaces` term in the spec is used for tabs as well. Will add test case and fix the code. Thanks!

Reply:
addressed incl. tests